torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -663,7 +663,7 @@ Tensor evenly_distribute_backward(Tensor grad, const Tensor & input, const Tenso
     return mask * (grad / mask.sum());
   } else {
     auto mask = value.isnan().item<bool>() ? input.isnan() : input == value;
-    return at::zeros_like(input).masked_fill_(mask, grad / mask.sum());
+    return grad.new_zeros(input.sizes(), input.options()).masked_fill_(mask, grad / mask.sum());
```

Comment:
It makes sense that the batch info are taken from grad and the other sizes from input.
I think it is worth mentioning in the "vmap gotcha" (if you have that) that the `new_*` functions behave this way.

Reply:
There isn't a vmap gotcha section anywhere, but I'll make a note on that in the future